libcfnet/protocol.h

```diff
@@ -0,0 +1,40 @@
+/*
+   Copyright 2019 Northern.tech AS
+
+   This file is part of CFEngine 3 - written and maintained by Northern.tech AS.
+
+   This program is free software; you can redistribute it and/or modify it
+   under the terms of the GNU General Public License as published by the
+   Free Software Foundation; version 3.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
+
+  To the extent this program is licensed as part of the Enterprise
+  versions of CFEngine, the applicable Commercial Open Source License
+  (COSL) may apply to this file if you as a licensee so wish it. See
+  included file COSL.txt.
+*/
+
+#ifndef CFENGINE_PROTOCOL_H
+#define CFENGINE_PROTOCOL_H
+
+#include <cfnet.h>
+#include <sequence.h>
+
+/**
+ * Creates a sequence from a directory listing from the server.
+ *
+ * @param [in] conn  Connection to use.
+ * @param [in] path  Path to retrieve directory listing of
+ * @return A list
```

Comment:
*A list* is not really too descriptive. :smile:

Reply:
Heh, I should have added WIP to this. I still need to add it in `cf-net` as well.